sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java

```diff
@@ -0,0 +1,234 @@
+
+				Log.i("SORMAS_LBDS", "Process LbdsPropagateKexToSormasIntent..");
+
+				Log.i("SORMAS_LBDS", "Sormas public key: " + kexToSormasIntent.getSormasKey());
+
+				PublicKey lbdsKey = kexToSormasIntent.getLbdsKey();
+				ConfigProvider.setLbdsServicePublicKey(lbdsKey);
+				Log.i("SORMAS_LBDS", "Lbds public key: " + lbdsKey);
+
+				String aesSecret = kexToSormasIntent.getAesSecret(ConfigProvider.getLbdsSormasPrivateKey());
+				ConfigProvider.setLbdsAesSecret(aesSecret);
+				Log.i("SORMAS_LBDS", "Lbds AES secret: " + Base64.encodeToString(aesSecret.getBytes(StandardCharsets.UTF_8), Base64.DEFAULT));
+
+				break;
+			default:
+				Log.i("SORMAS_LBDS", "unknown LBDS intent");
+			}
+		}
+		Log.i("SORMAS_LBDS", "==========================");
+	}
+
+	private void processLbdsResponsePersons(HttpContainer httpContainerResponse) {
+
+		HttpMethod methodFromResponse = httpContainerResponse.getMethod();
+		HttpResult resultFromResponse = httpContainerResponse.getResult();
+
+		Log.i("SORMAS_LBDS", "Request: " + methodFromResponse);
+		Log.i("SORMAS_LBDS", "Result Headers: " + resultFromResponse.headers);
+		Log.i("SORMAS_LBDS", "Result Body: " + resultFromResponse.body);
+
+		Gson gson = RetroProvider.initGson();
+		List<PersonDto> sentPersonDtos = gson.fromJson(methodFromResponse.payload, new TypeToken<List<PersonDto>>() {
+		}.getType());
+		List<PushResult> responsePushResults = gson.fromJson(resultFromResponse.body, new TypeToken<List<PushResult>>() {
+		}.getType());
+		Log.i("SORMAS_LBDS", "sent " + sentPersonDtos.size() + " PersonDtos and received " + responsePushResults.size() + " PushResults");
+
+		if (sentPersonDtos.size() != responsePushResults.size()) {
+			Log.i("SORMAS_LBDS", "List lengths differ, abort.");
+			throw new IllegalArgumentException("Number of sent Dtos and received PushResults must be equal");
+		}
+
+		for (int i = 0; i < sentPersonDtos.size(); i++) {
+			PersonDto personDto = sentPersonDtos.get(i);
+			String uuid = personDto.getUuid();
+			PushResult pushResult = responsePushResults.get(i);
+			if (pushResult != PushResult.OK) {
+				Log.i("SORMAS_LBDS", "Ignore PushResult " + pushResult + " for PersonDto " + uuid);
+				continue;
+			}
+			PersonDao personDao = DatabaseHelper.getPersonDao();
+			Person person = personDao.queryUuid(uuid);
+			if (person == null) {
+				Log.i("SORMAS_LBDS", "Person " + uuid + " not found, skip.");
+				continue;
+			}
+			if (!person.isModified()) {
+				Log.i("SORMAS_LBDS", "Person " + uuid + " not modified, skip.");
+				continue;
+			}
+			Person snapshot = personDao.querySnapshotByUuid(uuid);
+			if (snapshot != null) {
+				Log.i("SORMAS_LBDS", "Person " + uuid + " has a snapshot, skip.");
+				continue;
+			}
+			try {
+				if (!LbdsDtoHelper.isModifiedLbds(person, personDto, true)) {
+					personDao.accept(person);
+				} else {
+					// TODO: create snapshot
```

Comment:
This should be implemented...

Reply:
Refactored to only send new cases / persons that have not been sent via LBDS yet.